### PR TITLE
feat(stack): Adds new `Stack` component to arrange content and actions

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #6743 #5664

## Summary

- Adds new component `Stack` to render a row of actions and content.
- Can be used to slot into places where no layout options are present.
- Can be used to disable specific parts of slotted content.

## Background

This will help users be able to layout content that is slotted within a panel footer for example.

- The panel footer slot doesn't offer a way to slot things at the start or end and this would allow them the ability to do that without having to dig into CSS or apply padding.

It will also allow a use to disable specific parts within the component if necessary since some parent components don't allow disabling partial UI.

- Instead of disabling the component, the user would use this component and slot it into the component. The user can disable only a portion of slotted content inside this component.

## Questions

- Name of this component

### CODE REVIEW

File: src/components/layout-item/layout-item.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
+:host {
```
Comment (macandcheese) [inline src/components/layout-item/layout-item.scss:1]: Should we consider a margin variable or other spacing modifier like we have on input or rely on the content placed inside?

File: src/components/layout-item/layout-item.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+/**
+ * @slot - A slot for adding content.
+ * @slot actions-start - A slot for adding actionable `calcite-action` elements before the content of the component.
+ * @slot content-start - A slot for adding non-actionable elements before content of the component.
+ * @slot content-end - A slot for adding non-actionable elements after content of the component.
+ * @slot actions-end - A slot for adding actionable `calcite-action` elements after the content of the component.
+ */
+@Component({
+  tag: "calcite-layout-item",
```
Comment (macandcheese) [inline src/components/layout-item/layout-item.tsx:14]: Is this going to be public or internal component? The “-item” portion of the name makes me think there would be a parent of some kind - layout / layout-group.

File: src/components/layout-item/layout-item.scss [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  @apply flex flex-col flex-1;
+}
+
+.container {
```
Comment (macandcheese) [inline src/components/layout-item/layout-item.scss:5]: Should we consider a bordered option or rely on users to add that if needed?

File: src/components/layout-item/usage/Basic.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+```html
```
Comment (macandcheese) [inline src/components/layout-item/usage/Basic.md:1]: For best practices, what is the distinction between using this and using List with selection mode none?

File: src/components/layout-item/layout-item.scss (hunk 3 above)
Comment (driskull) [reply src/components/layout-item/layout-item.scss:5]: I think we can rely on users or where this is being slotted to handle that.

**Comment by driskull** [comment]
@macandcheese @jcfranco can we deprecate `detachedHeightScale` in favor of just `heightScale` and then in the doc we can say it will only work for `displayMode="float"`? That would give us freedom to add other displayModes where this scale may apply in the future

File: src/components/layout-item/layout-item.tsx (hunk 2 above)
Comment (macandcheese) [reply src/components/layout-item/layout-item.tsx:14]: ‘calcite-content-row’ / ‘layout-row’ (if we don’t expect it to have a “column” layout) ?

File: src/components/layout-item/layout-item.scss (hunk 1 above)
Comment (driskull) [reply src/components/layout-item/layout-item.scss:1]: We can include padding by default and allow a user to remove it with a variable if that sounds good?

File: src/components/layout-item/layout-item.tsx (hunk 2 above)
Comment (driskull) [reply src/components/layout-item/layout-item.tsx:14]: Good question. We can always just start with row layout and not have column until the need arises.

**Comment by driskull** [comment]
@macandcheese @jcfranco I looked at other web component libraries to find a name for this component and the closest seems to be `stack`.

The other closest I saw was `slats`. https://ux.mailchimp.com/patterns/slats

We could also go with something more generic like:

- box
- section
- group

Let me know what you think so I can get the name finalized.

**Comment by driskull** [comment]
Can I get a review plz :) ✨

File: src/components/stack/stack.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  shadow: true
+})
+export class Stack {
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**  When `true`, content interaction is prevented and the displayed with lower opacity. */
```
Comment (jcfranco) [inline src/components/stack/stack.tsx:24]: Typo: `prevented and the displayed`

File: src/components/stack/stack.stories.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { modesDarkDefault } from "../../../.storybook/utils";
+import readme from "./readme.md";
+import { html } from "../../../support/formatting";
+import { storyFilters } from "../../../.storybook/helpers";
+
+export default {
+  title: "Components/Stack",
```
Comment (jcfranco) [inline src/components/stack/stack.stories.ts:7]: WDYT about adding this under `Components/Support/` to better convey it's purpose for now?

**Review by jcfranco — APPROVED** [review]
💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲
💲💲💵💵💵💲💵💵💵💵💵💲💵💵💵💵💲💲💵💵💵💲💵💲💲💵💲💲💵💵💵💲
💲💵💲💲💲💲💲💲💵💲💲💲💵💲💲💵💲💵💲💲💲💲💵💲💵💲💲💵💲💲💲💲
💲💲💵💵💲💲💲💲💵💲💲💲💵💵💵💵💲💵💲💲💲💲💵💵💲💲💲💲💵💵💲💲
💲💲💲💲💵💲💲💲💵💲💲💲💵💲💲💵💲💵💲💲💲💲💵💲💵💲💲💲💲💲💵💲
💲💵💵💵💲💲💲💲💵💲💲💲💵💲💲💵💲💲💵💵💵💲💵💲💲💵💲💵💵💵💲💲
💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲💲

Awesome! This component is going to be extremely useful.

**Pull request merged**